performance/src/efficiency/finding_performance_writer.cpp

```diff
@@ -171,18 +171,31 @@ void finding_performance_writer::write_common(
 
         if (!evt_data.m_found_meas_to_ptc_map.empty()) {
             particle_hit_counts = identify_contributing_particles(
-                measurements, evt_data.m_found_meas_to_ptc_map);
+                found_measurements, evt_data.m_found_meas_to_ptc_map);
         } else {
             particle_hit_counts = identify_contributing_particles(
-                measurements, evt_data.m_meas_to_ptc_map);
+                found_measurements, evt_data.m_meas_to_ptc_map);
         }
 
+        const auto major_ptc = particle_hit_counts.at(0).ptc;
+        const auto n_major_hits = particle_hit_counts.at(0).hit_counts;
+
+        // Truth measureemnt from the particle
+        const std::vector<measurement> truth_measurements =
+            evt_data.m_ptc_to_meas_map.at(major_ptc);
+
         // Consider it being matched if hit counts is larger than the half
         // of the number of measurements
-        assert(measurements.size() > 0u);
-        if (particle_hit_counts.at(0).hit_counts / measurements.size() >
-            m_cfg.matching_ratio) {
-            const auto pid = particle_hit_counts.at(0).ptc.particle_id;
+        assert(found_measurements.size() > 0u);
+        assert(truth_measurements.size() > 0u);
+        const bool reco_matched =
+            n_major_hits / found_measurements.size() > m_cfg.matching_ratio;
+        const bool truth_matched =
+            n_major_hits / truth_measurements.size() > m_cfg.matching_ratio;
```

Comment:
I am getting **very** worried about this code. 😦 Is `n_major_hits` a floating point number? I don't think it is...

But this division should of course be done at floating point precision. We cannot divide two integers. And now I'm getting worried about this type of issue possibly being present in other places as well. Even though the efficiency plots have been okay-looking. 🤔

Reply:
I agree this is worrying, but shouldn't `-Wconversion` pick up on this?

Reply:
EDIT: Nope, we only use `-Wfloat-conversion`. I would have sworn someone added `-Wconversion` at some point...

Reply:
Damn - there was bug in track classification since long. You are right. FIxed now